### DISCUSSION TITLE
manually setting image tag in case of alphabetical/numerical image policy

### DISCUSSION THREAD

**Discussion by singh-navpreet** [fluxcd/flux2#1715] [Q&A] [answered]
Hi, 
In some scenarios , we would like to do manual deployments/ image tag update. This can be achieved by setting semver like : 
```
flux create image policy podinfo \
--image-ref=podinfo \
--select-semver=5.0.0
```
[https://fluxcd.io/docs/guides/image-update/#revert-image-updates](url)

but can we do same in case of alphabetical /numerical image policy? where tags for images are not in semver syntax.
something like:

```
flux create image policy podinfo \
--image-ref=podinfo \
--select-alpha=master_21xxx
```
or 
```
flux create image policy podinfo \
--image-ref=podinfo \
--select-numeric=20210811.1
```

**Comment by stefanprodan** [comment, marked as answer]
We'll have to add an explicit field in the API for tag pinning, see https://github.com/fluxcd/image-reflector-controller/issues/117

**Reply by stefanprodan** [reply]
To answer your question, pinning is possible by changing the manifest to:
```yaml
kind: ImagePolicy
spec:
  filterTags:
    pattern: "20210811.1"
  policy:
    alphabetical:
      order: asc
```

**Reply by singh-navpreet** [reply]
thanks alot @stefanprodan  . this worked perfectly.

**Reply by venkatesh-mb** [reply]
How can we manage alphanumeric tag?
we have tags like this ex: 4judm4 & branch-0.4.3-jemc8idm.

How can we manage global level, just consider any new tag should be considered.

**Reply by kingdonb** [reply]
Flux today requires sortable tags. https://fluxcd.io/docs/guides/sortable-image-tags/

If you can put a semantically sortable string alphanumeric or numeric component into your image tags, then they can be used with Flux's image automation. If you cannot put it, you can still use all of the core features of Flux without image update automation.

There is further elaboration on this topic in the Flux v1 FAQ, why sortable image tags are necessary – tl;dr it is because the metadata cannot be retrieved in bulk but tag strings can be fetched in bulk, so it is orders of magnitude more efficient for Flux working this way: https://fluxcd.io/legacy/flux/faq/#flux-v1-runtime-behavior-doesnt-scale-well
